raven-android/src/main/java/com/getsentry/raven/android/event/helper/AndroidEventBuilderHelper.java

```diff
@@ -291,29 +290,51 @@ private static Boolean isCharging(Context ctx) {
      * @return the device's current kernel version, as a string (from uname -a)
      */
     private static String getKernelVersion() {
-        String errorMsg = "Exception while attempting to read kernel information";
-        BufferedReader br = null;
         try {
-            Process p = Runtime.getRuntime().exec("uname -a");
-            if (p.waitFor() == 0) {
-                br = new BufferedReader(new InputStreamReader(p.getInputStream()));
-                return br.readLine();
-            }
+            return formatKernelVersion(Util.readLine("/proc/version"));
         } catch (Exception e) {
-            Log.e(TAG, errorMsg, e);
-        } finally {
-            if (br != null) {
-                try {
-                    br.close();
-                } catch (IOException ioe) {
-                    Log.e(TAG, errorMsg, ioe);
-                }
-            }
+            Log.e(TAG, "Exception while attempting to read kernel information", e);
         }
 
         return null;
     }
 
+    private static String formatKernelVersion(String rawKernelVersion) {
+        // Example (see tests for more):
+        // Linux version 3.0.31-g6fb96c9 (android-build@xxx.xxx.xxx.xxx.com) \
+        //     (gcc version 4.6.x-xxx 20120106 (prerelease) (GCC) ) #1 SMP PREEMPT \
+        //     Thu Jun 28 11:02:39 PDT 2012
+
+        final String procVersionRegex =
+                "Linux version (\\S+) " + /* group 1: "3.0.31-g6fb96c9" */
+                "\\((\\S+?)\\) " +        /* group 2: "x@y.com" (kernel builder) */
+                "(?:\\(gcc.+? \\)) " +    /* ignore: GCC version information */
```

Comment:
Yes this is kernel only so gcc only for now but I know that Google is working on clang on kernel too, so this reg-exp might not be valid later, good catch ! Instead of blindly picking this code from Google as I did, I could just get the output of /proc/version if this is fine for you ? See a comment above for difference of output between uname and /proc/version.